internal/controller/thanosreceive_controller.go

```diff
@@ -184,13 +217,61 @@ func (r *ThanosReceiveReconciler) buildHashrings(receiver monitoringthanosiov1al
 			Retention:      string(*hashring.Retention),
 			StorageSize:    resource.MustParse(hashring.StorageSize),
 			ObjStoreSecret: objStoreSecret,
+			ExternalLabels: hashring.ExternalLabels,
 		}
 		opts = append(opts, opt)
 	}
 
 	return receive.BuildIngesters(opts)
 }
 
+// buildHashringConfig builds the hashring configuration for the ThanosReceive resource.
+func (r *ThanosReceiveReconciler) buildHashringConfig(ctx context.Context, receiver monitoringthanosiov1alpha1.ThanosReceive) (client.Object, error) {
+	logger := log.FromContext(ctx)
+
+	cm := &corev1.ConfigMap{}
+	err := r.Client.Get(ctx, client.ObjectKey{Namespace: receiver.GetNamespace(), Name: receiver.GetName()}, cm)
```

Comment:
Is embedded already I think
```suggestion
	err := r.Get(ctx, client.ObjectKey{Namespace: receiver.GetNamespace(), Name: receiver.GetName()}, cm)
```